src/game/SpellEffects.cpp

```diff
@@ -2927,37 +2941,6 @@ void Spell::EffectSummonPet(SpellEffectIndex eff_idx)
         return;
```

Comment:
The condition says 'else if (!NewSummon->LoadPetFromDB((Player*)m_caster, petentry))' - which means that we only skip this condition if we actually can summon a pet, and if we can summon a pet, why'd we want to delete it?